build-system/tasks/cherry-pick.js

```diff
@@ -103,65 +103,67 @@ function performCherryPick(sha) {
 }
 
 function cherryPick() {
-  const {push, remote = 'origin'} = argv;
-  const commits = (argv.commits || '').split(',').filter(Boolean);
-  let onto = String(argv.onto || '');
+  return new Promise((resolve, reject) => {
+    const {push, remote = 'origin'} = argv;
+    const commits = (argv.commits || '').split(',').filter(Boolean);
+    let onto = String(argv.onto || '');
 
-  if (!commits.length) {
-    log(red('ERROR:'), 'Must provide commit list with --commits');
-    process.exitCode = 1;
-    return;
-  }
-  if (!onto) {
-    log(red('ERROR:'), 'Must provide 13-digit AMP version with --onto');
-    process.exitCode = 1;
-    return;
-  }
-  if (onto.length === 15) {
-    log(
-      yellow('WARNING:'),
-      'Expected a 13-digit AMP version but got a 15-digit RTV;',
-      'ignoring channel prefix'
-    );
-    // Be forgiving if someone provides a version instead of a full RTV.
-    onto = onto.substr(2);
-  }
-  if (onto.length !== 13) {
-    log(red('ERROR:'), 'Expected 13-digit AMP version; got', cyan(onto));
-    process.exitCode = 1;
-    return;
-  }
+    if (!commits.length) {
+      const error = new Error('Must provide commit list with --commits');
+      error.showStack = false;
+      return reject(error);
```

Comment:
With [promises](https://javascript.info/promise-basics), it's enough to just call `reject()` or `resolve()` without having to `return` the results.

Reply:
This isn't to return the results but to be finished with the function. Otherwise continues to move through the function, resulting in something like this
```
gulp cherry-pick 
SUCCESS: Cherry-picked 0 commits onto release 
Using gulpfile ~/src/ampproject/amphtml/gulpfile.js
Starting 'cherry-pick'...
'cherry-pick' errored after 1.06 ms
Error: Must provide commit list with --commits
```

I could do `reject(); return;` but this way is simpler.

Reply:
Gotcha. My comment was hurriedly written, and didn't consider the early exit case. Thanks for pointing it out.